src/target/TargetCollection.php

```diff
@@ -30,15 +32,17 @@ class TargetCollection implements TargetInterface
 
     public function __construct(array $targets)
     {
-        $this->targets = $targets;
         $ids = [];
         $types = [];
         $states = [];
         foreach ($targets as $target) {
-            if ($target) {
+            if ($target instanceof TargetInterface) {
                 $ids[] = $target->getId();
                 $types[] = $target->getType();
                 $states[] = $target->getState();
+                $this->targets[] = $target;
+            } else {
+                throw new RuntimeException('Target ' . $target . ' is incorrect');
```

Comment:
This library already has its own exception classes hierarchy. You should use one of them, or create a new, if appropriate does not exists